### ISSUE TITLE
Add dynamic NPC detection to replace the NPC list

### ISSUE THREAD

**Issue by ILikePlayingGames** [BiscuitDevelopment/SkyblockAddons#366]
This is also intended to fix #367.

**Comment by iHDeveloper** [comment]
Should the detection include the important NPCs only?

**Comment by iHDeveloper** [comment]
I was working on this issue. Before I knew @biscuut was assigned to it.
I managed to solve some of it so I wanted to post what I managed to do so far in order to help 😄 

I managed to detect the NPCs with their names. I didn't make the NPC list dynamic.

The name hologram is in the same position as the NPC.
The status hologram (e.g. click) is in the same position but with `y - 0.25`.
But for an unknown reason, the `World#getEntitiesWithinAABB` returns the status hologram in the list.

The detection is being triggered by `PlayerListener#onEntityEvent`.
```java
if (main.getUtils().isOnSkyblock()
    && entity instanceof EntityOtherPlayerMP
    && main.getUtils().getLocation() != Location.ISLAND
    && main.getUtils().getLocation() != Location.GUEST_ISLAND) {

    double x = entity.posX;
    double y = entity.posY;
    double z = entity.posZ;

    AxisAlignedBB axisAlignedBB = new AxisAlignedBB(x, y + 1D, z, x, y + 1D, z);
    for (EntityArmorStand hologram : entity.getEntityWorld().getEntitiesWithinAABB(EntityArmorStand.class, axisAlignedBB)) {
        if (!hologram.hasCustomName())
            continue;

        String name = hologram.getCustomNameTag();
        if (name.equals("§e§lCLICK"))
            continue;
        else if (name.equals("§6§lCOMING SOON"))
            continue;

        // TODO REMOVE DEBUG MESSAGE
        main.getUtils().sendMessage("§eDetect NPC §9" + entity.getName() + " §e-> §r" + name);
    }
}
```

# Showcases
- Random Place in Hub
![image](https://user-images.githubusercontent.com/20463031/90706093-10ee0a80-e29d-11ea-86d9-592f3ce405c7.png)

- Auction House
![image](https://user-images.githubusercontent.com/20463031/90706130-295e2500-e29d-11ea-9969-5cd6ba713c29.png)

**Comment by ILikePlayingGames** [comment]
Fixed in ef5ff08122679b60536cd993ea3fffc7e5569b13.
